### PR TITLE
Added automation link click statistics endpoint

### PR DESCRIPTION
**Pull request by troyciesco**
closes https://linear.app/ghost/issue/NY-1486

Automation analytics needs per-link unique click counts, but calculating them during normal automation reads would add unnecessary query cost. A dedicated endpoint lets Admin load link details only when requested.

Behavior:
- validates that the action belongs to the requested automation and is not deleted
- groups destinations by URL hash across every revision of the action
- counts distinct members, so repeat clicks and clicks across revisions are deduplicated
- includes zero-click links and orders results by click count, then URL

Review guide:
1. Start with getAutomationActionLinks for the aggregation and ownership boundary.
2. Follow browseActionLinks through the API controller and authenticated route.
3. Check useBrowseAutomationActionLinks for the lazy Admin client integration.
4. See the repository, API, and client tests for the expected contract and edge cases.

### CODE REVIEW

File: ghost/core/core/server/web/api/endpoints/admin/routes.js [hunk 1]
```diff
@@ -208,6 +208,7 @@
 
     // ## Automations
     router.get('/automations', mw.authAdminApi, http(api.automations.browse));
+    router.get('/automations/:automation_id/actions/:action_id/links', mw.authAdminApi, http(api.automationActionLinks.browse));
```
[the remark this answers is not in the document]
Comment (troyciesco) [reply ghost/core/core/server/web/api/endpoints/admin/routes.js:211]: False positive: this is an authenticated, permission-checked, read-only analytics endpoint. It performs no sensitive or externally observable action, and the adjacent limiter is specifically for sending test emails.

**Comment by troyciesco** [comment]
@coderabbitai review

**Review by cmraible — APPROVED** [review]
LGTM!

**Review by cmraible — CHANGES_REQUESTED** [review]
Apologies for redacting my prior approval 😅

I *think* we need to do the link replacements in here somewhere to remove the `__GHOST_URL__` placeholders. I didn't notice until I checked out the frontend PR and saw this:

File: ghost/core/core/server/services/automations/database-automations-repository.ts [hunk 2]
```diff
@@ -165,6 +170,37 @@ export function createDatabaseAutomationsRepository({
...
+                .innerJoin('automation_action_revisions as revisions', 'revisions.id', 'redirects.automation_action_revision_id')
+                .leftJoin('members_click_events', 'members_click_events.redirect_id', 'redirects.id')
+                .where('revisions.action_id', actionId)
+                .whereNotNull('redirects.to_hash')
+                .groupBy('redirects.to_hash')
+                .orderBy('clicked_count', 'desc')
+                .orderBy('url', 'asc');
+
+            return rows.map(row => ({
+                url: urlUtils.transformReadyToAbsolute(row.url),
```
Comment (troyciesco) [inline ghost/core/core/server/services/automations/database-automations-repository.ts:199]: fixed the GHOST_URL issue here
before:
Image
after:
Image

i balked at seeing the full site url instead of just `/chicken` but seems like that's what we do for newsletters too (makes sense bc it's an email)

**Review by cmraible — APPROVED** [review]
LGTM!
